### DISCUSSION TITLE
How to write a graph classification model using torch_geometric.graphgym

### DISCUSSION THREAD

**Discussion by SongZan222** [pyg-team/pytorch_geometric#7393] [Q&A] [answered]
How to write a graph classification model using torch_geometric.graphgym？There are too few references on the Internet
I hope you can get help soon. Thank you！

**Comment by SongZan222** [comment]
I would like to have a code that compares multiple aggregations or pools

**Comment by rusty1s** [comment]
Did you take a look at https://github.com/pyg-team/pytorch_geometric/blob/master/graphgym/configs/pyg/example_graph.yaml?

In the end, you just need to ensure that `task: graph` is set, and everything else should be similar to any other GraphGym model. Let me know if you stumble upon any issues.

**Reply by SongZan222** [reply]
Yes, I found the file earlier, but when I tried to run run_single.sh, I got an error: register_act('swish', partial(SWISH, inplace=cfg.mem.inplace))
AttributeError: 'NoneType' object has no attribute 'mem'
I used to program in an IDE and wasn't comfortable with this scripting language

**Reply by rusty1s** [reply]
Seems like `cfg` is `None` for you. Can you run `pip install yacs`?

**Reply by SongZan222** [reply]
After I installed the two packages as prompted by the script, I get the error from train() :

TypeError: __init__() got an unexpected keyword argument 'auto_select_gpus'

**Reply by rusty1s** [reply]
Which version of PyTorch Lightning are you on?

**Reply by SongZan222** [reply]
pytorch-lightning             2.0.2
torch-geometric               2.2.0
above is my environment

**Reply by rusty1s** [reply]
I see. You may need to upgrade to PyG 2.3 for PyTorch Lightning 2.0 support.

**Reply by SongZan222** [reply, marked as answer]
thank，The script is now working properly！
and one more question,
Does torch_geometric.graphgym only support starting tasks with scripts like this?
Not in some python compiler (such as PyCharm)?

**Reply by rusty1s** [reply]
You could mimic the logic we have here to run it within your own Pythons script: https://github.com/pyg-team/pytorch_geometric/blob/master/graphgym/main.py
